### ISSUE TITLE
Inconsistent HTTP errors when requesting unknown Bitstream vs. Item

### ISSUE THREAD

**Issue by ybnd** [DSpace/DSpace#10099]
## Describe the bug
Noticed a discrepancy between how our Item and Bitstream endpoints handle unknown IDs:
- Random valid UUID that doesn't exist in the database
  - https://demo.dspace.org/server/api/core/items/f8606711-03e1-4c37-b687-e9b2278a25ad → HTTP 404
  - https://demo.dspace.org/server/api/core/bitstreams/f8606711-03e1-4c37-b687-e9b2278a25ad → **HTTP 401**
- Non-UUID String
  - https://demo.dspace.org/server/api/core/items/not-a-uuid → **HTTP 401**
  - https://demo.dspace.org/server/api/core/bitstreams/not-a-uuid → HTTP 404

As far as I know all of these should return HTTP 404
- The first case should fail when the database query misses
- The second cae should fail when no request matcher is found for that path pattern

I can't think of a reason why the Item and Bitstream endpoints should behave differently.

### Additional notes
- We don't explain any HTTP error codes for `/api/core/bitstreams/<:uuid>` [in the contract](https://github.com/DSpace/RestContract/blob/main/bitstreams.md#single-bitstream)
- This could be related to pre-authorization
- Other endpoints may be affected too

## To Reproduce
Compare the HTTP responses on the URLs listed above

## Expected behavior
Missing resources should result in HTTP 404. 
If we do expect HTTP 401 in these cases for some reason, this should be documented in the REST contract.

## Related work
N/A?

**Comment by tdonohue** [comment]
@ybnd : I agree, these should all return 404 if the object **doesn't exist**.  However, they obviously should all return 401 if the object **does exist** but is access restricted (and the user isn't logged in).

Pulling this over to the maintenance board as this definitely seems to be a bug.  Needs a volunteer

**Comment by Zsolt05** [comment]
Hey @tdonohue and @ybnd!

I've implemented a fix for this issue and am preparing the backend(DSpace/DSpace#12954) and REST Contract(DSpace/RestContract#375) pull requests.

The fix makes the Item and Bitstream endpoints consistently return `404 Not Found` for:

- malformed UUIDs;
- valid but nonexistent UUIDs;
- UUIDs belonging to another DSpace object type.

The existing authorization behavior for resources that do exist remains unchanged: anonymous users receive `401 Unauthorized`, while authenticated users without sufficient permissions receive `403 Forbidden`.

I also added integration tests for the reported cases and verified the relevant existing security regressions in the Docker-based test environment:

- 9 targeted integration tests passed;
- 0 failures or errors;
- Checkstyle passed with 0 violations.
